cli/command/image/push.go

```diff
@@ -47,7 +55,7 @@ func runPush(dockerCli command.Cli, remote string) error {
 	authConfig := command.ResolveAuthConfig(ctx, dockerCli, repoInfo.Index)
 	requestPrivilege := command.RegistryAuthenticationPrivilegedFunc(dockerCli, repoInfo.Index, "push")
 
-	if command.IsTrusted() {
+	if !opts.untrusted && dockerCli.IsTrusted() {
```

Comment:
Does this need to check both values? It seems like either we can remove `dockerCli.IsTrusted()` from the call to `AddTrustSigningFlags` or we can remove `dockerCli.IsTrusted()` from this line and  check only `opts.untrusted` . `dockerCli.IsTrusted()` is being used as the default value to the trust flag, so I don't think we need to use it twice.

same thing in push.